### DISCUSSION TITLE
Support Single-Drone Scenario, with Wildcard Dns

### DISCUSSION THREAD

**Discussion by t-moe** [jamsocket/plane#284] [Q&A] [answered]
Hi there,
This project looks like the perfect foundation for our use case:
We develop a webapp in .NET/Angular and want to spin up a backend (single docker container) for each developer when they click a link in the CI build output of our project (e.g. to Preview a PullRequest). 

We want to host plane on our azure infrastructure, but we don't need multiple drones right now. 
**Is there a way to disable the DNS stuff and just create a single drone and a scheduler?** 
We would then create a wildcard dns to map *.test-our-webapp.com => The-One-And-Only-Drone-IP. We really don't want to host multiple services (dns, scheduler, drone) for this, because we need to keep our cloud cost at a minimum....

Thanks in advance

**Comment by paulgb** [comment, marked as answer]
Hi Timo,

Yes, your use case should work.

- To disable the DNS server, remove the `[dns]` line from the toml file passed to the controller (e.g. [this line](https://github.com/drifting-in-space/plane/blob/0ab7a73a46aebd40f37d89146188c5a460fae9de/sample-config/plane-config/controller.toml#L9)).
- Since the DNS server is required for Plane to obtain its own certificates, you'll need to get certificates some other way and pass them in instead. Once you do, you can pass them in to the drone process with [drone.toml](https://github.com/drifting-in-space/plane/blob/0ab7a73a46aebd40f37d89146188c5a460fae9de/sample-config/plane-config/drone.toml#L70).

You'll still need three processes (nats server, controller, and drone), but they can all be run on the same machine (I recommend using Docker Compose for this).

Hope that helps, happy to elaborate more if that would be helpful.

**Comment by t-moe** [comment]
Hi Paul,
Thank you for fast response!
It works! Awesome!!

I've removed the firefox and dnsmasq containers from the docker-compose sample configuration, exposed the nats and drone https port to get the desired use case working. I've also built a small nodejs app that uses the NATS api to spawn our containers. Next, I will be looking at the authentication part...

**Comment by paulgb** [comment]
Awesome, glad that worked! I think we should provide an example like this out of the box, so I've opened #283.
